nshlib/nsh_init.c

```diff
@@ -110,10 +110,14 @@ void nsh_initialize(void)
   FAR struct console_stdio_s *pstate;
 #endif
 
+  /* populate NSH prompt string */
+
+  nsh_update_prompt();
+
 #if defined(CONFIG_NSH_READLINE) && defined(CONFIG_READLINE_TABCOMPLETION)
-  /* Configure the NSH prompt */
+  /* Configure readline prompt */
```

Comment:
should we call readline_prompt in nsh_update_prompt

Reply:
It seems no need as readline only holds the address, not a copy.

Reply:
OK